packages/common/solana/utils.ts

```diff
@@ -212,18 +251,24 @@ export async function sendAndConfirmTransaction(
     signature = bs58.encode(tx.signature!);
   }
 
+  if (!sendThrottler) {
+    sendThrottler = new PQueue({ concurrency: sendRate, intervalCap: 1, interval: 1000 });
+  }
+
   let blockheight = await connection.getBlockHeight(commitment);
   let transactionSent = false;
   const rawTransaction = tx.serialize();
   while (blockheight < hash.lastValidBlockHeight + 15) {
     try {
       if (blockheight < hash.lastValidBlockHeight || !transactionSent) {
-        await connection.sendRawTransaction(rawTransaction, {
-          maxRetries: 0,
-          minContextSlot: context.slot,
-          preflightCommitment: commitment,
-          skipPreflight: true,
-        });
+        await sendThrottler.add(async () =>
```

Comment:
Is it required for us to have global-like queue so we don't hit RPCs limit across several invocations of `send...` APi